core/server/common/src/main/webapp/js/browse.js

```diff
@@ -0,0 +1,222 @@
+/**
+ *
+ rebuild the browse page navigator pill like breadcrumbs.
+ *
+ @param {string} path current navigator path
+ *
+ */
+function rebuildNavPill(path) {
+  if (path.length > 0 && path.charAt(path.length - 1) === '/') {
+    path = path.substr(0, path.length - 1);
+  }
+  var pathSeg = path.split("/");
+  if (path == "/")
+    pathSeg.length = 1;
+  if (pathSeg != null && pathSeg.length > 0) {
+    pathSeg[0] = "/";
+    $("#pathUl").children().remove();
+    var curPath = "";
+    $.each(pathSeg, function (index, value) {
+      if (value.length === 0)
+        return;
+      curPath += value;
+
+      var liNode = document.createElement("li");
+      var aNode = document.createElement("a");
+      $(aNode).attr("href", "javascript:void(0)");
+      $(aNode).attr("onClick", 'generateData("' + curPath + '")');
+      $(aNode).append(value);
+      if (index == pathSeg.length - 1) {
+        $(liNode).attr("class", "active");
+      }
+      liNode.appendChild(aNode);
+      $("#pathUl").append(liNode);
+
+      if (value !== "/") {
+        curPath += "/"
+      }
+    })
+  }
+}
+
+function generateData(path) {
```

Comment:
this can be moved to the internal definition of the angular module so that there is no need for the global variable `globalScope`.